cedar-policy-generators/src/schema.rs

```diff
@@ -637,82 +638,64 @@ impl Schema {
-                                // The action applies to an unspecified principal
-                                // and resource, and no other entity types.
-                                None
-                            } else {
-                                Some(ApplySpec {
-                                    resource_types: pick_entity_types(&mut resource_types, u)?,
-                                    principal_types: pick_entity_types(&mut principal_types, u)?,
-                                    context: arbitrary_attrspec(&settings, &entity_type_names, u)?,
-                                })
-                            };
-                            if settings.enable_unspecified_apply_spec {
-                                apply_spec
-                            } else {
-                                match apply_spec {
-                                    Some(ApplySpec {
-                                        resource_types,
-                                        principal_types,
-                                        context,
-                                    }) if resource_types.is_some() || principal_types.is_some() => {
-                                        Some(ApplySpec {
-                                            resource_types: if resource_types.is_none() {
-                                                Some(vec![])
-                                            } else {
-                                                resource_types
-                                            },
-                                            principal_types: if principal_types.is_none() {
-                                                Some(vec![])
-                                            } else {
-                                                principal_types
-                                            },
-                                            context,
-                                        })
-                                    }
-                                    // `apply_spec` either is None or has both resource and principal types to be None
-                                    //  we fail early for these cases
-                                    _ => {
-                                        return Err(Error::NoValidPrincipalOrResourceTypes);
-                                    }
+                            let mut picked_resource_types =
+                                pick_entity_types(&mut resource_types, u)?;
+                            let mut picked_principal_types =
+                                pick_entity_types(&mut principal_types, u)?;
+                            // If we already have resource_types and principal_types, randomly make them empty
+                            if principal_and_resource_types_exist {
+                                if u.ratio(1, 8)? {
+                                    picked_resource_types = None;
```

Comment:
can we change this to `principal_resource_types.clear()`? @aaronjeline added this fix when unsupported was removed. I don't think it makes a difference but looks a bit cleaner.

Reply:
I think this would change what exactly the generator is doing. After aarons change, this can't be `None` so it we make it empty instead. Prior to that change, `None` and empty are different.

Reply:
Ah right, depends on whether you want to test unsupported or not. I'm guessing we should do both, maybe we can add two more coin flips for empty then?